piqe_ocp_lib/api/resources/ocp_health_checker.py

```diff
@@ -0,0 +1,318 @@
+
+        NOTE : If persistence storage is not configured for image registry doesn't mean unhealthy. It's a WARNING. If
+        persistence storage is not configure for image registry, Images will be inaccessible after reboot.
+        """
+        is_image_registry_storage_configured = False
+        is_management_state_correct = False
+        is_persistence_storage_configured = True
+        logger.info("Check managementState for image registry")
+        image_config_response = self.ocp_config.get_ocp_config(name="cluster")
+        if image_config_response["spec"]["managementState"] == "Managed":
+            is_management_state_correct = True
+        logger.info("Is managementState correct : %s", {is_management_state_correct})
+
+        logger.info("Check if persistence storage configured for image registry")
+        if "emptyDir" in image_config_response["spec"]["storage"]:
+            is_persistence_storage_configured = False
+        logger.info("Is persistence Storage Configured: %s", is_persistence_storage_configured)
+
+        if is_management_state_correct and is_persistence_storage_configured:
+            is_image_registry_storage_configured = True
+
+        return is_image_registry_storage_configured
+
+    def check_api_server_health(self):
+        """
+        Check openshift apiserver is reachable and  healthy
+        :return: (boolean) Return True if api server is healthy otherwise False
+        """
+        status_codes = dict()
+        is_api_server_healthy = False
+        kubeconfig_data = self.get_data_from_kubeconfig_v4()
+
+        logger.info("Check health of API Server")
+        api_server_url = kubeconfig_data["api_server_url"]
+        final_api_server_url = api_server_url + "/healthz"
+        logger.info("API Server URL : %s", final_api_server_url)
+        bearer_token = self.ocp_secret.get_long_live_bearer_token()
+        headers = {'Authorization': 'Bearer ' + bearer_token}
+        api_server_response = requests.get(final_api_server_url, headers=headers, verify=False)
+        logger.info("API Server Status Code : %s", api_server_response.status_code)
+        status_codes["api_server_status"] = api_server_response.status_code
+
+        for key, value in status_codes.items():
+            if value in range(200, 300):
+                is_api_server_healthy = True
+
+        return is_api_server_healthy
+
+    def check_web_console_health(self):
+        """
+        Check if web console is reachable and healthy
+        :return:(boolean) Return True if web console is healthy otherwise False
+        """
+        status_codes = dict()
+        is_web_console_healthy = False
+
+        logger.info("Check health of web-console")
+        web_console_route = self.ocp_route.get_route_in_namespace(namespace="openshift-console", route_name="console")
+        web_console_url = "https://" + web_console_route + ":443" + "/healthz"
```

Comment:
Is healthz a typo?

Reply:
No, healthz is api endpoint to check health of web console. Otherwise it will fail.